File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap17/XMLRPC_ClientForCGIServer.pl
#!/usr/bin/perl -w

### XMLRPC_ClientForCGIServer.pl

use Frontier::Client;
use strict;

my $server_url = 'http://localhost/cgi-bin/XMLRPC_CGIServerInPerl.cgi';    
                                                                     #(A)
my $server = Frontier::Client->new( url => $server_url );            #(B)
my $result = $server->call('ACMEWebService.sumAndDifference', 5, 3); #(C)
my $sum = $result->{'sum'};                                          #(D)
my $difference = $result->{'difference'};                            #(E)
print "Sum: $sum, Difference: $difference\n";                        #(F)
                                  # Sum: 3, Difference: 3
$result = $server->call('ACMEWebService.hello');                     #(G)
print "$result\n";                                                   #(H)
           # hello world from the XML-RPC CGI server in Perl

File: projects/engineering.purdue.edu/scriptingwithobjects/swocode/chap17/XMLRPC_ClientForCGIServer.py
#!/usr/bin/env python

### XMLRPC_ClientForCGIServer.py

import xmlrpclib

server_url = 'http://localhost/cgi-bin/XMLRPC_CGIServerInPython.cgi' #(A)
server = xmlrpclib.ServerProxy( server_url )                         #(B)

#print server.ACMEWebService.sumAndDifference( 5, 3 )
print server.sumAndDifference( 5, 3 )                                #(C)
                              # {'sum': 8, 'difference': 2}            
print server.sayHello()                                              #(D)
               # hello world from the Python CGI XML-RPC server
print server.bye()                                                   #(E)
               # goodbye world from the Python CGI XML-RPC server
